loopy/transform/precompute.py

```diff
@@ -280,6 +289,63 @@ class _not_provided:  # noqa: N801
     pass
 
 
+class FunctionNameCollector(CombineMapper):
+    def combine(self, values):
+        from functools import reduce
+        return reduce(frozenset.union, values, frozenset())
+
+    def map_call(self, expr):
+        return self.combine([frozenset([expr.function.name])]
+                            + [self.rec(arg) for arg in expr.parameters])
+
+    def map_call_with_kwargs(self, expr):
+        return self.combine([frozenset([expr.function.name])]
+                            + [self.rec(arg) for arg in expr.parameters]
+                            + [self.rec(arg) for arg in expr.kw_parameters.values()])
+
+    def map_algebraic_leaf(self, expr):
+        return frozenset()
+
+    def map_constant(self, expr):
+        return frozenset()
+
+
+@memoize_on_first_arg
+def _get_calls_in_expr(expr):
+    return FunctionNameCollector()(expr)
+
+
+@memoize_on_first_arg
+def get_all_deps_of_insn(insn):
```

Comment:
```suggestion
def _get_called_names(insn):
```
* and then move the `insn.read_dependency_names()` out into an `or` conditional, rather than explicitly constructing the union
* and maybe make a new predicate `contains_subst_rule_invocation(insn)`?

Reply:
Agreed, thanks! Fixed.